src/libraries/System.Text.Json/src/System/Text/Json/Serialization/ReadStack.cs

```diff
@@ -19,6 +19,21 @@ internal struct ReadStack
         private List<ReadStackFrame> _previous;
         public int _index;
 
+        // The bag of preservable references. It needs to be kept in the state and never in JsonSerializerOptions otherwise it may be affected by race conditions.
```

Comment:
"race conditions" does not really explain the reason -- the options should not have any per-serialization state since every serialization shares the same immutable state on the options. I'd just remove the last sentence of the comment.